sensors/lidar/lidar_test.go

```diff
@@ -0,0 +1,147 @@
+package lidar_test
+
+import (
+	"context"
+	"errors"
+	"image/color"
+	"testing"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/testutils/inject"
+	"go.viam.com/test"
+
+	"go.viam.com/slam/sensors/lidar"
+)
+
+const (
+	testLidarName  = "testLidarName"
+	wrongLidarName = "wrongLidarName"
+)
+
+func TestNew(t *testing.T) {
+	t.Run("Empty sensors array failure", func(t *testing.T) {
+		cam := &inject.Camera{}
+		deps := make(resource.Dependencies)
+		deps[camera.Named(testLidarName)] = cam
+		sensors := []string{}
+		sensorIndex := 0
+
+		testLidar, err := lidar.New(deps, sensors, sensorIndex)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldEqual, "index out of bounds")
+		test.That(t, testLidar, test.ShouldResemble, lidar.Lidar{})
+	})
+
+	t.Run("Out of bounds index failure", func(t *testing.T) {
```

Comment:
Suggestion since there are several test cases where the error is "index out of bounds"

```suggestion
	t.Run("Sensor index out of sensor array bounds failure", func(t *testing.T) {
```